File: apps/frontend/src/types/sponsor.ts
// Sponsor tier types
export type SponsorTier = "platinum" | "gold" | "silver" | "bronze";

// Sponsor interface - matches the expected backend structure
export interface Sponsor {
  id: string;
  name: string;
  logo: string; // URL or path to the sponsor logo image
  tier: SponsorTier;
  description: string;
  website: string;
  featured?: boolean;
  active?: boolean; // Whether the sponsor is currently active
  createdAt?: string;
  updatedAt?: string;
}

// API response types for when you connect to backend
export interface SponsorsResponse {
  sponsors: Sponsor[];
  total: number;
}

export interface SponsorDetailResponse {
  sponsor: Sponsor;
}

// Filter options for sponsors
export interface SponsorFilters {
  tier?: SponsorTier | "all";
  featured?: boolean;
  active?: boolean;
}

File: apps/backend/app/models/sponsor.py
from beanie import Document, Indexed
from pydantic import Field, HttpUrl, ConfigDict
from datetime import datetime
from typing import Optional
from enum import Enum


class SponsorTier(str, Enum):
    """Sponsor tier levels"""
    PLATINUM = "platinum"
    GOLD = "gold"
    SILVER = "silver"
    BRONZE = "bronze"


class Sponsor(Document):
    """Sponsor document model for MongoDB using Beanie ODM"""

    name: Indexed(str, unique=True)  # type: ignore
    tier: SponsorTier
    # Public URL to logo served via API; will be '/api/v1/sponsors/{id}/logo'
    logo: Optional[str] = None
    # GridFS file id for the stored logo
    logo_file_id: Optional[str] = None
    website: Optional[HttpUrl] = None
    description: Optional[str] = None
    featured: bool = False
    active: bool = True
    display_order: int = 0  # For controlling display order on frontend
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "sponsors"  # MongoDB collection name
        use_state_management = True
        indexes = [
            "name",
            "tier",
            [("display_order", 1)],
            [("created_at", -1)],
        ]

    def __repr__(self):
        return f"<Sponsor {self.name} ({self.tier})>"

    def __str__(self):
        return self.name
